src/gardenlinux/features/parser.py

```diff
@@ -217,6 +221,82 @@ def get_cname_as_feature_set(cname):
         cname = cname.replace("_", "-_")
         return set(cname.split("-"))
 
+    @staticmethod
+    def get_flavor_from_cname(cname: str, get_arch: bool = True) -> str:
+        """
+        Extracts the flavor from a canonical name.
+
+        This method parses a Garden Linux canonical name (cname) and extracts
+        the flavor component, with or without the architecture suffix.
+
+        Example canonical names:
+        - "aws-gardener_prod-amd64"
+        - "azure-gardener_prod_tpm2_trustedboot-amd64-1312.2-80ffcc87"
+
+        The flavor is the platform plus feature string (e.g., "aws-gardener_prod")
+
+        Args:
+            cname (str): Canonical name of an image
+            get_arch (bool): Whether to include the architecture in the returned flavor
+                            If True: returns "aws-gardener_prod-amd64"
+                            If False: returns "aws-gardener_prod"
+
+        Returns:
+            str: The extracted flavor string, with or without architecture
+        """
+        # Use regex to extract components from the canonical name
+        # This handles complex cnames with version and commit hash
+        re_match = re.match(
+            "([a-zA-Z0-9]+([\\_\\-][a-zA-Z0-9]+)*?)(-([a-z0-9]+)(-([a-z0-9.]+)-([a-z0-9]+))*)?$",
+            cname,
+        )
+
+        assert re_match, f"Not a valid GardenLinux canonical name {cname}"
+
+        # Split the cname to separate platform and features
+        parts = cname.split("-")
```

Comment:
That statement won't work as we also have a valid "cname" with `metal_pxe`. Not sure if the mixture of string operations and the regex above is needed at this stage or just makes the whole code over-complicated to read. You may just take the remaining RegEx logic from `__main__.py` to construct `cname_base` (flavor), `arch` and `version` out of the given input. It might be cleaner to add a class parsing and providing the result as properties:
```
class CName(object):
  _flavor = None
  _arch = None
  _version = None
  _commit_id = None

  def __init__(self, cname):
    ..parsing..

  @property
  def flavor(self):
    return self._flavor
  ...
```